From ee8fe4a771e9a33b8e6faa8f9352ff3831c2957e Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Mon, 6 Jan 2025 10:23:17 +0100
Subject: [PATCH] packagegroup-core-standalone-sdk-target: Add missing RDEPENDS
 to the swupdate SDK

The SDK for a custom image is not able to compile full featured swupdate,
because some static libs and headers are missing (liblua.a, ..).

corelib/lua_interface.c:20:10: fatal error: lua.h: No such file or directory
   20 | #include "lua.h"
      |          ^~~~~~~

Using TOOLCHAIN_TARGET_TASK += "lua-staticdev" in a custom image recipe
has the overhead of figuring out, what libs are relevant to build
swupdate. For easier development / meta-layer integration let's add the
missing dependencies via packagegroup-core-standalone-sdk-target.

The libs are now implicit added to TOOLCHAIN_TARGET_TASK via:
populate_sdk_base:
    TOOLCHAIN_TARGET_TASK += "packagegroup-core-standalone-sdk-target"

Upstream-Status: Submitted
[https://patchwork.ozlabs.org/project/swupdate/patch/20250106135329.188852-1-Michael.Glembotzki@iris-sensing.com/]

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 .../packagegroup-core-standalone-sdk-target.bbappend          | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 recipes-core/packagegroup/packagegroup-core-standalone-sdk-target.bbappend

diff --git a/recipes-core/packagegroup/packagegroup-core-standalone-sdk-target.bbappend b/recipes-core/packagegroup/packagegroup-core-standalone-sdk-target.bbappend
new file mode 100644
index 0000000..14ec963
--- /dev/null
+++ b/recipes-core/packagegroup/packagegroup-core-standalone-sdk-target.bbappend
@@ -0,0 +1,4 @@
+RDEPENDS:${PN} += "\
+    lua-staticdev \
+    mtd-utils-staticdev \
+"
-- 
2.47.1

